eventhubs/client/azure-eventhubs/src/test/java/com/azure/eventhubs/BackCompatTest.java

```diff
@@ -0,0 +1,132 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.eventhubs;
+
+import com.azure.core.amqp.Retry;
+import com.azure.core.util.logging.ClientLogger;
+import com.azure.eventhubs.implementation.ApiTestBase;
+import com.azure.eventhubs.implementation.ReactorHandlerProvider;
+import org.apache.qpid.proton.Proton;
+import org.apache.qpid.proton.amqp.Binary;
+import org.apache.qpid.proton.amqp.messaging.ApplicationProperties;
+import org.apache.qpid.proton.amqp.messaging.Data;
+import org.apache.qpid.proton.amqp.messaging.MessageAnnotations;
+import org.apache.qpid.proton.message.Message;
+import org.junit.Assert;
+import org.junit.Ignore;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.TestName;
+import reactor.core.publisher.Flux;
+import reactor.test.StepVerifier;
+
+import java.nio.ByteBuffer;
+import java.time.Duration;
+import java.util.HashMap;
+
+import static java.nio.charset.StandardCharsets.UTF_8;
+
+public class BackCompatTest extends ApiTestBase {
+    private static final String PARTITION_ID = "0";
+    private static final String APPLICATION_PROPERTY = "firstProp";
+    private static final String INT_APPLICATION_PROPERTY = "intProp";
+    private static final String PAYLOAD = "testmsg";
+
+    private final ClientLogger logger = new ClientLogger(BackCompatTest.class);
+
+    private EventHubClient client;
+    private EventHubProducer sender;
+    private EventHubConsumer receiver;
+    private ReactorHandlerProvider handlerProvider;
+
+    @Rule
+    public TestName testName = new TestName();
+
+    @Override
+    protected String testName() {
+        return testName.getMethodName();
+    }
+
+    @Override
+    protected void beforeTest() {
+        logger.asInfo().log("[{}]: Performing test set-up.", testName.getMethodName());
+
+        handlerProvider = new ReactorHandlerProvider(getReactorProvider());
+        client = new EventHubClient(getConnectionOptions(), getReactorProvider(), handlerProvider);
```

Comment:
It makes this test easier to read if you create the sender and receiver here. Otherwise, the sender/receiver do not need to be member variables of this class.